### PR TITLE
libbson: init at 1.1.10

### PR DESCRIPTION
**Pull request by iand675**
Adds the libbson package.

### CODE REVIEW

File: pkgs/development/libraries/libbson/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    rev = "refs/tags/${version}";
+    sha256 = "9ad2df5238cdb52397daac119f3647e25fdca9165333adb30282af984c15f03a";
+  };
+
+  buildInputs = [ autoconf automake libtool perl which ];
+
+  configureScript = "./autogen.sh";
+
+  meta = with stdenv.lib; {
+    description = "Provides useful routines related to building, parsing, and iterating BSON documents. It is a useful base for those wanting to write high-performance C extensions to higher level languages such as python, ruby, or perl.";
```
Comment (bjornfor) [inline pkgs/development/libraries/libbson/default.nix]: Please change "description" according to http://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes. Use "longDescription" (in addition to "description") if you want more text.

**Comment by bjornfor** [comment]
libbson readme file says that you don't need autogen.sh if building from tarball. I suggest trying the tarball. I guess that means fetchzip from github archive url.

**Comment by iand675** [comment]
Done

**Comment by bjornfor** [comment]
Then I guess "autoconf, automake114x," are unused and can be removed?

**Comment by iand675** [comment]
No, the build fails without them. The only thing using the tar release buys is not having to run autogen.sh, but the downside is that it requires using the automake114x package instead of just using automake.

**Comment by iand675** [comment]
Here's the error that I get without automake114x:

```
building
make flags: SHELL=/nix/store/zmd4jk4db5lgxb8l93mhkvr3x92g2sx2-bash-4.3-p39/bin/bash
CDPATH="${ZSH_VERSION+.}:" && cd . && /nix/store/zmd4jk4db5lgxb8l93mhkvr3x92g2sx2-bash-4.3-p39/bin/bash /tmp/nix-build-libbson-1.1.10.drv-0/libbson-1.1.10.tar.gz/build/autotools/missing aclocal-1.14 -I build/autotools/m4
/tmp/nix-build-libbson-1.1.10.drv-0/libbson-1.1.10.tar.gz/build/autotools/missing: line 81: aclocal-1.14: command not found
WARNING: 'aclocal-1.14' is missing on your system.
         You should only need it if you modified 'acinclude.m4' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'aclocal' program is part of the GNU Automake package:
         <http://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <http://www.gnu.org/software/autoconf>
         <http://www.gnu.org/software/m4/>
         <http://www.perl.org/>
Makefile:849: recipe for target 'aclocal.m4' failed
make: *** [aclocal.m4] Error 127
builder for ‘/nix/store/pq2g5wf5xrlskbxpvw1i8ab1lzw6wa10-libbson-1.1.10.drv’ failed with exit code 2
```

**Comment by bjornfor** [comment]
@iand675: Thanks for explaining. Pushed to master (fd7144e45514844641b030fb2bcf365b32959a32). I removed trailing dot in description and unneeded parenthesis around mkDerivation arguments.

**Pull request closed without merging**